src/MetaProperty.ts

```diff
@@ -31,6 +31,11 @@ export class MetaProperty {
      */
     public static readonly PAGE_MODEL_ROUTER = 'cq:pagemodel_router';
 
+    /**
+     * Meta property pointing to wcm mode
+     */
+    public static readonly WCM_MODE = 'cq:wcmmode';
```

Comment:
`aemmode` URL Query Param.

Reply:
Priority of mode selection : 
1. Meta Property.
2 . Query param

Reply:
can be now removed since we have `AEM_MODE_KEY` in `Constants`

Reply:
the value of these 2 constants are different, isn't it? One is the name of the query parameter. The other one is the name of the meta property. Names are different. And we cannot rename the meta property without introducing a breaking change for the consumers. `cq:wcmmode` seems pretty uncommon for the name of a query parameter

Reply:
Yes, that is why I am hesitant in removing this Meta property. Moreover going by the consistency of this property across the code I feel it makes it easier to understand.